wandb/sdk/launch/agent/agent.py

```diff
@@ -112,7 +112,17 @@ def _job_is_scheduler(run_spec: Dict[str, Any]) -> bool:
         return False
 
     if run_spec.get("resource") == "local-process":
-        # If a scheduler is a local-process (100%), also
+        # If a scheduler is a job, and run in local-process, check
+        # whitelist of supported wandb jobs
+        job = run_spec.get("job")
+        if job and job.split(":")[0].split("/")[-1] in [
+            "job-WandbScheduler",
+            "job-OptunaScheduler",
+            "job-HyperoptScheduler",
```

Comment:
nit: store these in an enum, also we'll want a way to not count custom schedulers as well...